src/server_manager/web_app/ui_components/outline-gcp-create-server-app.ts

```diff
@@ -303,32 +307,55 @@ export class GcpCreateServerApp extends LitElement {
     } else {
       if (!this.billingAccounts || this.billingAccounts.length === 0) {
         this.showBillingAccountSetup();
+        // Check every five seconds to see if an account has been added.
+        this.billingAccountsRefreshLoop = window.setInterval(() => {
+          this.refreshBillingAccounts();
+        }, 5000);
       } else {
         this.showProjectSetup(this.project);
       }
     }
   }
 
+  disconnectedCallback() {
```

Comment:
Does this component actually get disconnected? Some components are just hidden. It depends on the parent

Reply:
Currently it does not, so this code is just for future-proofing or contractual correctness.

Reply:
Isn't this a leak?
Doesn't that mean that you will keep polling forever?
And keep adding new loops every time you open the create server screen?

Reply:
Perhaps we can check the visibility in the loop somehow. I don't remember how we do in the other parts of the app.

Reply:
Reopening the create server screen will call `start()` on the existing element.  `start()` stops any existing polling loop via `this.init()`.  The loop also stops once you add an account.

It's true that the loop will continue if you hide the GCP "app" and never add a billing account.

Reply:
Here is what you can do:
- Add a stop() method
- In app-root.js, add a handler for currentPage changes that will call the stop method.

This is exactly what we do in the outline-server-view to manage the animation:
https://github.com/Jigsaw-Code/outline-server/blob/e86c1981ed9ec8ce4d96be2bd088b73b8dbfad48/src/server_manager/web_app/ui_components/outline-server-view.js#L973

It's possible to set a on-changed handler on the iron-page, or a property observer on currentPage.